src/IdentityServer/Endpoints/PushedAuthorizationEndpoint.cs

```diff
@@ -84,12 +84,17 @@ public async Task<IEndpointResult> ProcessAsync(HttpContext context)
 
         var response = await _responseGenerator.CreateResponseAsync(parValidationResult.ValidatedRequest);
 
-        return response switch
+        switch(response)
         {
-            PushedAuthorizationSuccess success => new PushedAuthorizationResult(success),
-            PushedAuthorizationFailure fail => new PushedAuthorizationErrorResult(fail),
-            _ => throw new Exception("Can't happen")
-        };
+            case PushedAuthorizationSuccess success:
+                Telemetry.Metrics.PushedAuthorizationRequest(parValidationResult.ValidatedRequest.Client.ClientId);
+                return new PushedAuthorizationResult(success);
+            case PushedAuthorizationFailure fail:
+                Telemetry.Metrics.PushedAuthorizationRequestFailure(parValidationResult.ValidatedRequest.ClientId, fail.Error);
+                return new PushedAuthorizationErrorResult(fail);
+            default:
+                throw new Exception("Can't happen");
```

Comment:
@josephdecock maybe while reviewing this, you can consider a better exception message here?

Reply:
Yeah, good idea - this is done